app/com/microsoft/azure/iotsolutions/iothubmanager/services/DeviceProperties.java

```
@@ -296,17 +333,13 @@ private void parseWhitelist(List<String> whitelist,
     /**
      * @summary Get current device properties saved in cosmosDB cache
      */
-    private ValueApiModel getCurrentDevicePropertiesFromStorage() throws ExternalDependencyException {
+    private ValueApiModel getCurrentDevicePropertiesFromStorage() throws
+        ExternalDependencyException, ResourceNotFoundException, ConflictingResourceException {
         try {
             return this.storageClient.getAsync(CacheCollectionId, CacheKey).toCompletableFuture().get();
```

we want to throw resource not found exception now?

---

we are throwing it here and log it in updateListAsync method. So, it doesn't make any difference.